language-examples/python/vectara-rest/rest_api_key_queries.py

```diff
@@ -53,8 +53,19 @@ def query(customer_id: int, corpus_id: int, query_address: str, api_key: str, qu
                        response.reason,
                        response.text)
         return response, False
-    return response, True
 
+    message = response.json()
+    if (message["status"] and
+        any(status["code"] != "OK" for status in message["status"])):
+        logging.error("Query failed with status: %s", message["status"])
+        return message["status"], False
+
+    for response_set in message["responseSet"]:
+        for status in response_set["status"]:
+            if status["code"] != "OK":
+                return status, False
```

Comment:
This and previous block shows how complicated and granular our API is. v2 will be much cleaner.

Reply:
+1